Fix/background health check

baejooc opened the pull request:
## Title
Fix health check bugs: ensure latest model list, correct interval type, and prevent infinite loop

<!-- e.g. "Implement user authentication feature" -->

## Relevant issues

<!-- e.g. "Fixes #000" -->
Fixes https://github.com/BerriAI/litellm/issues/10114


## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🐛 Bug Fix
🧹 Refactoring

## Changes
- Deep copy llm_model_list on each health check iteration to ensure latest changes are used.
- Fix type inconsistency for health_check_interval (float → int).
- Prevent infinite loop when the 'health_check_interval' is unset or invalid.

baejooc committed:
fix: improve health check logic by deep copying model list on each iteration

baejooc committed:
test: add async test for background health check reflecting model list changes

baejooc committed:
fix: validate health check interval before executing background health check

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=10887) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=10887) before we can accept your contribution.<br/><hr/>**baejooc** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=10887) it.</sub>

krrishdholakia commented:
Hi @baejooc can you rebase with main - i believe the testing pipeline should be stable now. 

Also 

> proxy/proxy_server.py:836: error: Need type annotation for "health_check_results" (hint: "health_check_results: dict[<type>, <type>] = ...")  [var-annotated]


looks like you need to add this typing

baejooc committed:
fix: specify type for health check results dictionary

baejooc commented:
@krrishdholakia 
I’ve rebased with main now that the testing pipeline has stabilized, and I’ve also added a type hint for health_check_results.
Let me know if there's anything else I should update.
